### PR TITLE
refactor: use define_benchmarks macro

### PR DESCRIPTION
**Pull request by gregdhill**
De-duplicates between `list_benchmark` and `add_benchmark`, also uses `WhitelistedStorageKeys`

### CODE REVIEW

File: parachain/runtime/interlay/src/lib.rs [hunk 1]
```diff
@@ -1834,3 +1832,28 @@ cumulus_pallet_parachain_system::register_validate_block! {
...
+
+    #[test]
+    fn check_whitelist() {
+        let whitelist: HashSet<String> = AllPalletsWithSystem::whitelisted_storage_keys()
+            .iter()
+            .map(|e| HexDisplay::from(&e.key).to_string())
+            .collect();
+
+        // Block Number
+        assert!(whitelist.contains("26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac"));
```
Comment (daniel-savu) [inline parachain/runtime/interlay/src/lib.rs:1851]: Shouldn't `Total Issuance` also be included in these checks?

File: parachain/runtime/testnet-interlay/src/lib.rs [hunk 2]
```diff
@@ -1419,39 +1431,20 @@ impl_runtime_apis! {
...
-            add_benchmark!(params, batches, replace, Replace);
-            add_benchmark!(params, batches, vault_registry, VaultRegistry);
-            add_benchmark!(params, batches, dex_general, DexGeneral);
-            add_benchmark!(params, batches, dex_stable, DexStable);
-            add_benchmark!(params, batches, dex_swap_router, DexSwapRouter);
-
-            if batches.is_empty() { return Err("Benchmark not found for this pallet.".into()) }
+            add_benchmarks!(params, batches);
             Ok(batches)
         }
```
Comment (daniel-savu) [inline parachain/runtime/testnet-interlay/src/lib.rs:1449]: Shouldn't we also add unit tests to check the whitelist in the testnet runtimes?

**Review by daniel-savu — APPROVED** [review]

File: parachain/runtime/testnet-interlay/src/lib.rs (hunk 2 above)
Comment (gregdhill) [reply parachain/runtime/testnet-interlay/src/lib.rs:1449]: Not worth it, we will remove those runtimes soon

File: parachain/runtime/interlay/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply parachain/runtime/interlay/src/lib.rs:1851]: No because we don't use `pallet-balances`

**Pull request merged**